src/com/reason/ide/template/OCamlContentType.java

```diff
@@ -13,9 +16,51 @@ protected OCamlContentType() {
         super("OCAML", "Ocaml");
     }
 
+    protected OCamlContentType(String name) {
+        super("OCAML."+name, name, OCamlContentType.class);
+    }
+
     @Override
     public boolean isInContext(@NotNull TemplateActionContext templateActionContext) {
         final String name = templateActionContext.getFile().getName();
         return name.endsWith(".ml") || name.endsWith(".mli");
     }
+
+    // Code-only
+    private static final class OCamlCodeTemplates extends ScopeTemplates {
+        public OCamlCodeTemplates() {
+            super("Code", false);
+        }
+    }
+    // Comments only
+    private static final class OCamlCommentTemplates extends ScopeTemplates {
+        public OCamlCommentTemplates() {
+            super("Comment", true);
+        }
+    }
+
+    private static class ScopeTemplates extends OCamlContentType {
+        private final boolean comments;
```

Comment:
code style is using `my` prefix for class variables and not using `this.`

Reply:
Oh! I wondered why I had a "my" before the name! I will fix it, thanks!